[clang][www] Documentation revision and proof read for features.html

lknknm opened the pull request:
This is mostly just a small proof read for the https://clang.llvm.org/features.html documentation page. 
The changes include some typo fixes and suggestions I've found would be useful.
If needed, I can also proof read other sections of the documentation and pack everything into a "bigger" PR to do everything in one go.

lknknm committed:
Fix: Change Reference to 'BSD' license to 'Apache' License.

Commit:
Fix: 'a optimizer' to 'an optimizer' typo

Commit:
Fix(suggestion): 'Fast compiles' to 'Fast Compile-times'

Commit:
Suggestion: change to 'when you begin implementing a language'

lknknm committed:
Fix: 'libraries and tool' to 'libraries and tools'

lknknm committed:
Fix(suggestion): 'As an example of the power of this library based design'

lknknm committed:
Fix(plural rule): IDE's to IDEs

lknknm committed:
Merge branch 'main' into Documentation-Revision-and-Proof-Read-for-Features.html

DavidSpickett commented on clang/www/features.html:
```diff
@@ -28,7 +28,7 @@ <h1>Clang - Features and Goals</h1>
 
 <ul>
-<li><a href="#performance">Fast compiles and low memory use</a></li>
+<li><a href="#performance">Fast Compile-times and low memory use</a></li>
```
"fast compile-times" is probably correct English but reads awkwardly for me, and is a bit on the informal side for technical documentation.

Instead you could change "compiles" to "compilation". This is a simpler overall.

DavidSpickett commented on clang/www/features.html:
```diff
@@ -183,7 +183,7 @@ <h3><a name="libraryarch">Library Based Architecture</a></h3>
 </ul>
 
-<p>As an example of the power of this library based design....  If you wanted to
+<p>As some examples for the power of this library based design: If you wanted to
```
This started as an awkward sentence, I see what you did to repair it but perhaps rephrasing is in order. How about:
"Some examples of the power of this library based design are:"

If you want to keep your version, at least use "of" instead of "for".
"As some examples of the power..."

lknknm replied:
Yes, mostly I just wanted to replace one of the `of` with another term/word to avoid repetition "of the... of this", but I think overall it's fine, I'll go with your suggestion as "Some examples of the power of this library based design are:"

DavidSpickett reviewed (commented):
Thanks for taking the time to contribute.

A couple of comments, but everything else makes sense.

DavidSpickett commented:
I don't see the BSD to Apache 2 change in the final changes. Did you realise it was incorrect? Perhaps it was accidentally removed in a later commit.

lknknm committed:
Fix: change "compile-times" term to "compilation"

lknknm committed:
Fix: "As some examples..." to "Some examples..."

lknknm committed:
Fix: 'BSD' to 'Apache 2' license link

lknknm commented:
> I don't see the BSD to Apache 2 change in the final changes. Did you realise it was incorrect? Perhaps it was accidentally removed in a later commit.

Yes, I think something might have happened as I changed git local accounts, but now I've corrected and implemented all the suggested changes :) The one who realized it was incorrect was @myxogast at Discord and pointed out so I could correct everything on the same PR.

> This is fine. I think you were probably told to include a bit more basically to show "good faith" in contributing, and I see that here so I'm ok with the scope of this.

Regarding this, yes :) I was reading the documentation and saw a few typos so I've got recommended to proof-read the whole text so we could address multiple things under the same PR. In the future I think I can contribute with some other things, but I still need more technical knowledge on the matter. I'll stick around anyways though, as sometimes I look into the Discord threads :)

lknknm committed:
Merge branch 'main' into Documentation-Revision-and-Proof-Read-for-Features.html

DavidSpickett reviewed (approved):
LGTM.

You need to adjust your GitHub email settings before I can merge this. See https://llvm.org/docs/DeveloperPolicy.html#email-addresses.
